File: app/src/main/java/com/typ/muslim/integrations/telegram/models/TelegramRequestResult.java
/*
 * This app is developed by AHMED SLEEM
 *
 * Copyright (c) 2021.  TYP INC. All Rights Reserved
 */

package com.typ.muslim.integrations.telegram.models;

import com.typ.muslim.managers.AManager;

import org.json.JSONObject;

import java.util.HashMap;
import java.util.Map;

public class TelegramRequestResult {

	private final boolean succeed;
	private final Map<String, Object> result;

	public TelegramRequestResult(boolean succeed, Map<String, Object> result) {
		this.succeed = succeed;
		this.result = result == null ? new HashMap<>() : result;
		AManager.log("TelegramRequestResult", toString());
	}

	public boolean hasSucceed() {
		return succeed;
	}

	public boolean hasResult() {
		return result != null && !result.isEmpty();
	}

	public int optInt(String key, int fallback) {
		if (result != null) {
			Object value = result.get(key);
			return value != null ? (int) value : fallback;
		}
		return fallback;
	}

	public boolean optBoolean(String key, boolean fallback) {
		if (result != null) {
			Object value = result.get(key);
			return value != null ? (boolean) value : fallback;
		}
		return fallback;
	}

	public String optString(String key, String fallback) {
		if (result != null) {
			Object value = result.get(key);
			return value != null ? (String) value : fallback;
		}
		return fallback;
	}

	public JSONObject optJSONObject(String key, JSONObject fallback) {
		if (result != null) {
			Object value = result.get(key);
			return value != null ? (JSONObject) value : fallback;
		}
		return fallback;
	}

	@Override
	public String toString() {
		return "TelegramRequestResult{" +
		       "hasSucceed=" + succeed +
		       ", result=" + result +
		       '}';
	}

}

File: app/src/main/java/com/typ/muslim/features/telegram/models/TelegramRequestResult.kt
/*
 * This app is developed by AHMED SLEEM
 *
 * Copyright (c) 2021.  TYP INC. All Rights Reserved
 */
package com.typ.muslim.features.telegram.models

import org.json.JSONObject

class TelegramRequestResult(
    private val succeed: Boolean,
    result: Map<String?, Any?>?
) {
    private val result: Map<String, Any>? = null

    fun hasSucceed(): Boolean = succeed

    fun hasResult(): Boolean = result != null && result.isNotEmpty()


    fun optInt(key: String, fallback: Int): Int {
        if (result != null) {
            val value = result[key]
            return if (value != null) value as Int else fallback
        }
        return fallback
    }

    fun optBoolean(key: String, fallback: Boolean): Boolean {
        if (result != null) {
            val value = result[key]
            return if (value != null) value as Boolean else fallback
        }
        return fallback
    }

    fun optString(key: String, fallback: String): String {
        if (result != null) {
            val value = result[key]
            return if (value != null) value as String else fallback
        }
        return fallback
    }

    fun optJSONObject(key: String, fallback: JSONObject): JSONObject {
        if (result != null) {
            val value = result[key]
            return if (value != null) value as JSONObject else fallback
        }
        return fallback
    }

    override fun toString(): String {
        return "TelegramRequestResult{" +
                "hasSucceed=" + succeed +
                ", result=" + result +
                '}'
    }
}